src/parser/basic.rs

```diff
@@ -0,0 +1,45 @@
+//! Default [`Parser`] implementation.
+
+use std::{path::Path, vec};
+
+use futures::stream;
+
+use crate::Parser;
+
+/// Default [`Parser`].
+///
+/// As there is no async runtime-agnostic way to interact with io, this
+/// [`Parser`] is blocking.
+#[derive(Clone, Copy, Debug)]
+pub struct Basic;
+
+impl<I: AsRef<Path>> Parser<I> for Basic {
+    type Output = stream::Iter<vec::IntoIter<gherkin::Feature>>;
+
+    fn parse(self, path: I) -> Self::Output {
+        let path = path
+            .as_ref()
+            .canonicalize()
+            .expect("failed to canonicalize path");
+
+        let features = if path.is_file() {
+            let env = gherkin::GherkinEnv::default();
```

Comment:
Do not be lazy with imports.

`gherkin::GherkinEnv` is bad. `GherkinEnv` is good.  
`gherkin::Feature` is good.  
`globwalk::GlobWalkerBuilder` is bad. `GlobWalkerBuilder` is good.

The rule is simple: do not repeat the same word twice in-place.